ui/src/shared/actions/sources.ts

```diff
@@ -220,29 +225,34 @@ export const getSourcesAsync = () => async (dispatch): Promise<Source[]> => {
 
 const addSourceLink = async (source: Source): Promise<Source> => {
   try {
+    if (!validateFluxVersion(source)) {
+      return source
+    }
+
     const services = await getServicesAJAX(source.links.services)
     const fluxServices = services.filter(s => s.type === 'flux')
 
     let service: Service
-
-    if (validateSupportsFlux(source)) {
-      if (!fluxServices.length) {
-        service = await createFluxServiceForSource(source)
+    if (!fluxServices.length) {
+      service = await createFluxServiceForSource(source)
+    } else {
+      if (fluxServices[0].url !== source.url) {
```

Comment:
this `if / else` block can be flattened as an `else if` and `else` clause in the parent if / else block

Reply:
```typescript
let service: Service

if (!fluxServices.length) {
  service = await createFluxServiceForSource(source)
} else if (fluxServices[0].url !== source.url) {
  service = await updateFluxService(source, fluxServices[0])
} else {
  service = fluxServices[0]
}
```

Reply:
sounds good!